src/compiler/builder.ts

```diff
@@ -384,7 +386,12 @@ namespace ts {
             // Get next batch of affected files
             if (!state.currentAffectedFilesSignatures) state.currentAffectedFilesSignatures = new Map();
             if (state.exportedModulesMap) {
-                if (!state.currentAffectedFilesExportedModulesMap) state.currentAffectedFilesExportedModulesMap = new Map();
+                if (!state.currentAffectedFilesExportedModulesMap) {
+                    state.currentAffectedFilesExportedModulesMap = {
+                        exporting: BuilderState.createTwoWayMap<Path, Path>(),

```

Comment:
Nit: Do you really need type parameters?

Reply:
I've made it a specialized type.  The main downside has been losing enforced separation between keys and values.